File: packages/i18nify-js/src/modules/currency/__tests__/formatIndianNumber.test.ts
import formatIndianNumber from '../formatIndianNumber';

describe('formatIndianNumber', () => {
  it('formats 1 lakh (1,00,000)', () => {
    expect(formatIndianNumber(100000)).toBe('1,00,000');
  });

  it('formats 10 lakhs (10,00,000)', () => {
    expect(formatIndianNumber(1000000)).toBe('10,00,000');
  });

  it('formats 1 crore (1,00,00,000)', () => {
    expect(formatIndianNumber(10000000)).toBe('1,00,00,000');
  });

  it('formats number with decimals', () => {
    expect(formatIndianNumber(1234567.89)).toBe('12,34,567.89');
  });

  it('formats number below 1000 (no grouping needed)', () => {
    expect(formatIndianNumber(999)).toBe('999');
  });

  it('formats 1000 with grouping', () => {
    expect(formatIndianNumber(1000)).toBe('1,000');
  });

  it('formats 10000 with Indian grouping', () => {
    expect(formatIndianNumber(10000)).toBe('10,000');
  });

  it('formats 100000 with Indian grouping', () => {
    expect(formatIndianNumber(100000)).toBe('1,00,000');
  });

  it('formats zero', () => {
    expect(formatIndianNumber(0)).toBe('0');
  });

  it('formats negative number', () => {
    expect(formatIndianNumber(-1000000)).toBe('-10,00,000');
  });

  it('accepts string amount', () => {
    expect(formatIndianNumber('100000')).toBe('1,00,000');
  });

  it('throws for non-numeric string', () => {
    expect(() => formatIndianNumber('abc')).toThrow();
  });

  it('formats with INR currency', () => {
    const result = formatIndianNumber(100000, { currency: 'INR' });
    expect(result).toContain('1,00,000');
    expect(result).toContain('₹');
  });
});

File: packages/i18nify-go/modules/currency/format_indian_number_test.go
package currency

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestFormatIndianNumber(t *testing.T) {
	tests := []struct {
		name    string
		amount  interface{}
		want    string
		wantErr bool
	}{
		{"1 lakh", 100000, "1,00,000", false},
		{"10 lakhs", 1000000, "10,00,000", false},
		{"1 crore", 10000000, "1,00,00,000", false},
		{"below 1000", 999, "999", false},
		{"1000", 1000, "1,000", false},
		{"10000", 10000, "10,000", false},
		{"12,34,567", 1234567, "12,34,567", false},
		{"zero", 0, "0", false},
		{"negative 10 lakhs", -1000000, "-10,00,000", false},
		{"string input", "100000", "1,00,000", false},
		{"invalid string", "abc", "", true},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result, err := FormatIndianNumber(tt.amount, NumberFormatOptions{})
			if tt.wantErr {
				assert.Error(t, err)
				return
			}
			assert.NoError(t, err)
			assert.Equal(t, tt.want, result)
		})
	}
}

func TestFormatIndianNumber_WithCurrency(t *testing.T) {
	result, err := FormatIndianNumber(100000, NumberFormatOptions{Currency: "INR"})
	assert.NoError(t, err)
	assert.Contains(t, result, "1,00,000")
	assert.Contains(t, result, "₹")
}
